### PR TITLE
[DOC] Add header_file doxygen command.

### LINKED ISSUE #1010: Model header include for variables as a Doxygen Macro
Currently we use this snippet to document the include header for the view variable definitions.

https://github.com/seqan/seqan3/blob/cfea39c4248adeef490b491778231dc32c654507/include/seqan3/range/views/convert.hpp#L33-L36

However, this seems to be a problem for all typedefs and vairable definitions in the library.
I am wondering if we can add a  macro that generates this. Maybe even a functional one, that can generate the path from `__FILE__` macro?

### PR DESCRIPTION
**Pull request by smehringer**
Resolves #1010

Look and feel:

grafik

### CODE REVIEW

**Comment by smehringer** [comment]
I have no clue why the coverage changes fails :roll_eyes:

File: include/seqan3/range/views/complement.hpp [hunk 1]
```diff
@@ -30,12 +30,9 @@ namespace seqan3::views
  * \returns             A range of converted elements. See below for the properties of the returned range.
  * \ingroup views
  *
- * Calls seqan3::nucleotide_alphabet::complement() on every element of the input range.
+ * \header_file{seqan3/range/views/complement.hpp}
```
Comment (rrahn) [inline include/seqan3/range/views/complement.hpp]:
```suggestion
 * \details
 * 
 * \header_file{seqan3/range/views/complement.hpp}
```

File: include/seqan3/range/views/char_to.hpp [hunk 2]
```diff
@@ -32,10 +32,7 @@ namespace seqan3::views
  * \returns             A range of converted elements. See below for the properties of the returned range.
  * \ingroup views
  *
- * **Header**
- * ```cpp
- *      #include <seqan3/range/views/char_to.hpp>
- * ```
+ * \header_file{seqan3/range/views/char_to.hpp}
```
Comment (rrahn) [inline include/seqan3/range/views/char_to.hpp]:
```suggestion
 * \details
 * 
 * \header_file{seqan3/range/views/char_to.hpp}
```

File: include/seqan3/range/views/convert.hpp [hunk 3]
```diff
@@ -30,10 +30,7 @@ namespace seqan3::views
  * \returns             A range of converted elements. See below for the properties of the returned range.
  * \ingroup views
  *
- * **Header**
- * ```cpp
- *      #include <seqan3/range/views/convert.hpp>
- * ```
+ * \header_file{seqan3/range/views/convert.hpp}
```
Comment (rrahn) [inline include/seqan3/range/views/convert.hpp]:
```suggestion
 * \details
 * 
 * \header_file{seqan3/range/views/convert.hpp}
```

File: include/seqan3/range/views/get.hpp [hunk 4]
```diff
@@ -31,10 +31,7 @@ namespace seqan3::views
                         See below for the properties of the returned range.
  * \ingroup views
  *
- * **Header**
- * ```cpp
- *      #include <seqan3/range/views/get.hpp>
- * ```
+ * \header_file{seqan3/range/views/get.hpp}
```
Comment (rrahn) [inline include/seqan3/range/views/get.hpp]:
```suggestion
 * \details
 * 
 * \header_file{seqan3/range/views/get.hpp}
```

File: include/seqan3/range/views/rank_to.hpp [hunk 5]
```diff
@@ -31,10 +31,7 @@ namespace seqan3::views
  * \returns             A range of converted elements. See below for the properties of the returned range.
  * \ingroup views
  *
- * **Header**
- * ```cpp
- *      #include <seqan3/range/views/rank_to.hpp>
- * ```
+ * \header_file{seqan3/range/views/rank_to.hpp}
```
Comment (rrahn) [inline include/seqan3/range/views/rank_to.hpp]:
```suggestion
 * \details
 * 
 * \header_file{seqan3/range/views/rank_to.hpp}
```

File: include/seqan3/range/views/to_char.hpp [hunk 6]
```diff
@@ -30,10 +30,7 @@ namespace seqan3::views
  * \returns             A range of converted elements. See below for the properties of the returned range.
  * \ingroup views
  *
- * **Header**
- * ```cpp
- *      #include <seqan3/range/views/to_char.hpp>
- * ```
+ * \header_file{seqan3/range/views/to_char.hpp}
```
Comment (rrahn) [inline include/seqan3/range/views/to_char.hpp]:
```suggestion
 * \details
 * 
 * \header_file{seqan3/range/views/to_char.hpp}
```

File: include/seqan3/range/views/to_lower.hpp [hunk 7]
```diff
@@ -31,10 +31,7 @@ namespace seqan3::views
  * \returns             A range of converted elements. See below for the properties of the returned range.
  * \ingroup views
  *
- * **Header**
- * ```cpp
- *      #include <seqan3/range/views/to_lower.hpp>
- * ```
+ * \header_file{seqan3/range/views/to_lower.hpp}
```
Comment (rrahn) [inline include/seqan3/range/views/to_lower.hpp]:
```suggestion
 * \details
 * 
 * \header_file{seqan3/range/views/to_lower.hpp}
```

File: include/seqan3/range/views/to_rank.hpp [hunk 8]
```diff
@@ -30,10 +30,7 @@ namespace seqan3::views
  * \returns             A range of converted elements. See below for the properties of the returned range.
  * \ingroup views
  *
- * **Header**
- * ```cpp
- *      #include <seqan3/range/views/to_rank.hpp>
- * ```
+ * \header_file{seqan3/range/views/to_rank.hpp}
```
Comment (rrahn) [inline include/seqan3/range/views/to_rank.hpp]:
```suggestion
 * \details
 * 
 * \header_file{seqan3/range/views/to_rank.hpp}
```

File: include/seqan3/range/views/to_upper.hpp [hunk 9]
```diff
@@ -31,10 +31,7 @@ namespace seqan3::views
  * \returns             A range of converted elements. See below for the properties of the returned range.
  * \ingroup views
  *
- * **Header**
- * ```cpp
- *      #include <seqan3/range/views/to_upper.hpp>
- * ```
+ * \header_file{seqan3/range/views/to_upper.hpp}
```
Comment (rrahn) [inline include/seqan3/range/views/to_upper.hpp]:
```suggestion
 * \details
 * 
 * \header_file{seqan3/range/views/to_upper.hpp}
```

File: include/seqan3/range/views/translate.hpp [hunk 10]
```diff
@@ -774,10 +771,7 @@ namespace seqan3::views
  *
  * This view can be used to translate nucleotide sequences into aminoacid sequences (see translation_frames for possible combination of frames).
  *
- * **Header**
- * ```cpp
- *      #include <seqan3/range/views/translate.hpp>
- * ```
+ * \header_file{seqan3/range/views/translate.hpp}
```
Comment (rrahn) [inline include/seqan3/range/views/translate.hpp]: please, moved it up directly under \details.

**Pull request merged**